src/hooks/useToggle.ts

```diff
@@ -0,0 +1,21 @@
+import { useState } from "react";
+
+
+const useToggle =(defaultValue:boolean)=>{
+
+
+     const [value, setValue] = useState<boolean>(defaultValue);  
+
+
+    const toggleFunction= ():void =>{
+        
+        setValue(!value)     
```

Comment:
It would be better to use this syntax instead 
```tsx
setValue(prevVal=>!prevVal)
```
if the current state is false and we use `toggleFunction` two times after another
```tsx
toggleFunction();
toggleFunction();
```
The expected result would be false again, but it will be true not false because of react patching states and doing changes asynchronously.

The syntax provided will make sure that we read the last state value.